packages/core/src/VNodes/AtomicNode.ts

```diff
@@ -102,13 +102,13 @@ export class AtomicNode extends VNode {
         return undefined;
     }
     /**
-     * See {@link VNode.descendants}.
+     * See {@link AbstractNode.descendants}.
      *
      * @return Returns an empty array since an atomic node cannot have children.
      */
-    descendants<T extends VNode>(predicate?: Predicate<T>): [];
-    descendants<T>(predicate?: Predicate<T>): [];
-    descendants<T>(predicate?: Predicate<T>): [] {
+    descendants<T extends VNode>(predicate?: Predicate<T>): VNode[];
+    descendants<T>(predicate?: Predicate<T>): VNode[];
+    descendants<T>(predicate?: Predicate<T>): VNode[] {
```

Comment:
There is a limitation with what you can write as a type. If you write `VNode[]` then it gives no information on the length of the array. If you write `[]` like I used to do, it's an empty tuple, which is technically correct, but now Typescript does not know that these are VNodes, so writing this:
```typescript
const children = atomic.children();
children.indexOf(node):
```
Results in a typeing error because the type of `children` is `[]` so the type of the parameter of `indexOf` is `never`, which is true in some way, since there is no element that is going to fit in there. However, it's kind of sad that I can't just call my `indexOf` and get my `-1` as a result. From an other point of view, if typescript is able to realize that the node is an `AtomicNode` and not merely any `VNode`, it might actually be interesting that I stumble upon this typing warning because there is no point in calling `indexOf` and getting `-1` since the children already will *always* be empty. I'll change it back to `[]` and see what happens.

Reply:
Yes I think it can be a good thing to get that typing error but on the other hand not if it means having to check all the time if the node is atomic where you could just do nothing if `indexOf` returns `-1`.

Reply:
If it's a `VNode`, so not narrowed down to `AtomicNode`, `indexOf` will work as expected.